jmix-security/security-flowui/src/main/java/io/jmix/securityflowui/authentication/LoginViewSupport.java

```diff
@@ -308,7 +305,12 @@ protected Location getRedirectLocation(VaadinServletRequest request, VaadinServl
                 //and RouteConfiguration.getRoute(String) doesn't support full URLs
                 //like one returned from savedRequest.getRedirectUrl()
                 QueryParameters queryParameters = QueryParameters.fromString(defaultSavedRequest.getQueryString());
-                return new Location(defaultSavedRequest.getServletPath(), queryParameters);
+                String servletPath = defaultSavedRequest.getServletPath();
+
+                if (getRouteConfiguration().isPathAvailable(removeLeadingSlash(servletPath))) {
```

Comment:
Are you sure that `isPathAvailable` accepts `null`s? Probably it'd better to create own `isPathAvailable` method in which check `null` and only then `getRouteConfiguration().isPathAvailable`.

Reply:
Fixed